src/WebJobs.Script.WebHost/Swagger/SwaggerDocumentManager.cs

```diff
@@ -0,0 +1,168 @@
+        {
+            var apiEndpoints = new Dictionary<string, Dictionary<string, HttpOperationInfo>>();
+            foreach (var httpRoute in httpFunctions.Keys)
+            {
+                if (httpFunctions[httpRoute].Metadata.IsDisabled == true)
+                {
+                    continue;
+                }
+
+                string endpoint = $"/{httpRoute.RouteTemplate}";
+                Dictionary<string, HttpOperationInfo> endpointsOperationData;
+                if (!apiEndpoints.TryGetValue(endpoint, out endpointsOperationData))
+                {
+                    endpointsOperationData = new Dictionary<string, HttpOperationInfo>();
+                    apiEndpoints.Add(endpoint, endpointsOperationData);
+                }
+
+                string[] httpOperations = GetHttpOperations(httpRoute);
+                Collection<HttpOperationParameterInfo> inputParameters = GetInputParameters(httpRoute);
+                foreach (var httpOperation in httpOperations)
+                {
+                    // Trace is not recogized as an HttpMethod by swagger specification
+                    if (httpOperation.ToLower(CultureInfo.CurrentCulture) == HttpMethod.Trace.ToString().ToLowerInvariant())
+                    {
+                        continue;
+                    }
+
+                    HttpOperationInfo httpOperationInfo = new HttpOperationInfo()
+                    {
+                        InputParameters = inputParameters,
+                        OperationId = $"{endpoint}/{httpOperation}"
+                    };
+                    endpointsOperationData.Add(httpOperation, httpOperationInfo);
+                }
+            }
+            return apiEndpoints;
+        }
+
+        private static string[] GetHttpOperations(IHttpRoute httpRoute)
+        {
+            HttpMethodConstraint httpMethodConstraint = null;
+            httpRoute.Constraints.TryGetValue(ScriptConstants.HttpMethodConstraintName, out httpMethodConstraint);
+            if (httpMethodConstraint != null && httpMethodConstraint.AllowedMethods != null)
+            {
+                return httpMethodConstraint.AllowedMethods.Select(p => p.Method).ToArray<string>();
+            }
+            return ScriptConstants.HttpMethods.ToArray();
+        }
+
+        private static Collection<HttpOperationParameterInfo> GetInputParameters(IHttpRoute httpRoute)
+        {
+            var parameters = new Collection<HttpOperationParameterInfo>();
+            var pathParameters = GetAllPathParameters(httpRoute.RouteTemplate);
+            foreach (var constraint in httpRoute.Constraints)
+            {
+                var httpConstraint = constraint.Value as IHttpRouteConstraint;
+                SwaggerDataType? swaggerDataType = httpConstraint.ToSwaggerDataType();
+                if (swaggerDataType != null && httpConstraint.GetType() != typeof(HttpMethodConstraint))
```

Comment:
So the parameter won't be added if we can't resolve the data type? Should it still be added with an unknown type to give users the ability to set just that, instead of having to create the parameter (or miss the fact that it wasn't created)?